girderformindlogger/external/fix_flankers.py

```diff
@@ -0,0 +1,126 @@
+# Update flanker activities from old to new protocol
+# Old: https://raw.githubusercontent.com/mtg137/Flanker_applet/master/protocols/flanker/flanker_schema
+# New: https://raw.githubusercontent.com/ChildMindInstitute/mindlogger-flanker-applet/master/protocols/flanker/flanker_schema
+
+from girderformindlogger.models.item import Item
+from girderformindlogger.models.activity import Activity
+from girderformindlogger.models.folder import Folder
+from girderformindlogger.models.user import User
+from girderformindlogger.models.cache import Cache
+from girderformindlogger.utility import jsonld_expander
+from bson.objectid import ObjectId
+
+
+def prepare_items(activityId):
+    items = Item().find(query={'meta.activityId': activityId, 'meta.screen.@type.0': 'reprolib:schemas/Field'}, fields= {"_id": 1})
+    itemsCount = items.count()
+    print('total', itemsCount)
+    skipUntil = None
+    affectedActivityIds = []
+    for index, itemId in enumerate(items, start=1):
+        if skipUntil == itemId['_id']:
+            skipUntil = None
+        if skipUntil is not None:
+            continue
```

Comment:
I guess the part with `skipUntil` should not work.
You assign it to None and then you go through the items list.
Then you are comparing it with the `itemId`. It means you always compare it with `None`. Also, it means that you will never reach the point when the skipUntil is not `None`

Also, the naming in the loop is not correct. I think as far as you are iterating through the items list, the element should be called `item`

Overall, I guess it should be something like this:

```python
def prepare_items(activityId):
    items = Item().find(query={'meta.activityId': activityId, 'meta.screen.@type.0': 'reprolib:schemas/Field'}, fields= {"_id": 1})
    itemsCount = items.count()

    affectedActivityIds = []

    for index, item in enumerate(items, start=1):
        item_id = item["_id"]

        if not item_id:
            continue

        item = Item().findOne(item_id)
        print('processing', item['_id'], index, '/', itemsCount)
        affectedActivityIds.append(item['meta']['activityId'])
```
What do you think?

Reply:
I removed skipUntil, it's not needed in prepare_items